.github/workflows/checks.yml

```diff
@@ -0,0 +1,113 @@
+name: Checks
+
+on: workflow_call
+
+jobs:
+
+  version-check-job:
+    name: Version Check
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: SCM Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.5.0
+
+      - name: Check Version(s)
+        run: poetry run version-check version.py
+
+  build-documentation-job:
+    name: Build Documentation
+    needs: [version-check-job]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: SCM Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.5.0
+
+      - name: Build Documentation
+        run: |
+          poetry run python -m nox -s build-docs
+
+  lint-job:
+    name: Linting (Python-${{ matrix.python-version }})
+    needs: [version-check-job]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9"]
+
+    steps:
+      - name: SCM Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.5.0
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Tests
+        run: poetry run nox -s lint
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: .lint.txt
+          path: .lint.txt
+
+  type-check-job:
+    name: Type Checking (Python-${{ matrix.python-version }})
+    needs: [version-check-job]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9"]
+
+    steps:
+      - name: SCM Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.5.0
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Tests
```

Comment:
```suggestion
      - name: Run Type Check
```

Reply:
See comment above. I propose to create a ticket in github.com/exasol/python-toolbox/